lib/mistral_ai_summary_example/mistralai_summary_page.dart

```diff
@@ -1,13 +1,165 @@
+        actions: [
+          IconButton(
+            onPressed: () async {
+              final newSettings = await Navigator.push(
+                context,
+                MaterialPageRoute<SummarySettings>(
+                  builder: (BuildContext context) => SettingsWidget(
+                    mistralAIClient: mistralAIClient,
+                    initialSettings: summarySettings,
+                  ),
+                ),
+              );
+              setState(() {
+                summarySettings = newSettings ?? summarySettings;
+              });
+            },
+            icon: const Icon(Icons.settings),
+          ),
+        ],
+      ),
+      body: SafeArea(
+        child: Padding(
+          padding: const EdgeInsets.all(16),
+          child: SingleChildScrollView(
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: <Widget>[
+                TextField(
+                  decoration: const InputDecoration(
+                    labelText: 'Enter text to summarize',
+                  ),
+                  minLines: 1,
+                  maxLines: 10,
+                  onChanged: (value) => summaryInputController.text = value,
+                ),
+                const SizedBox(height: 20),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    ElevatedButton(
+                      onPressed: () {
+                        summarizeText(summaryInputController.text);
+                      },
+                      child: const Text('Summarize'),
+                    ),
+                    const SizedBox(width: 20),
+                    Column(
+                      children: [
+                        ElevatedButton(
+                          onPressed: () {
+                            summarizeText(summaryTextSample);
+                          },
+                          child: const Text('Summarize sample tex'),
+                        ),
+                        const SizedBox(height: 20),
+                        TextButton(
+                          onPressed: () => showDialog<String>(
+                            context: context,
+                            builder: (context) =>
+                                const SampleTextDialogButton(),
+                          ),
+                          child: const Text('See sample text'),
+                        ),
+                      ],
+                    ),
+                  ],
+                ),
+                const SizedBox(height: 20),
+                const Text('Summarized Text:'),
+                const SizedBox(height: 10),
+                SelectableText(summaryResult),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class SampleTextDialogButton extends StatelessWidget {
```

Comment:
WDYT about adding a `copy all text button` or at least `select all text button`? It will be easier to use it like that.

Reply:
Good idea, added in [6175491](https://github.com/nomtek/flutter_ai_examples/pull/3/commits/6175491ca6f1db066119f6262f5742a40e440234)